pyiron_gpl/pacemaker/pacemaker.py

```diff
@@ -0,0 +1,231 @@
+from pyiron_base import GenericJob, GenericParameters
+from pyiron_base.settings.generic import Settings
+from shutil import copyfile
+
+
+s = Settings()
+
+try:
+    from pyace import BBasisConfiguration, ACEBBasisSet
+
+    HAS_PYACE = True
+except ImportError as e:
+    print("Could not import `pyace` package. The package should be installed for proper functionality")
+    HAS_PYACE = False
+
+# set loggers
+loggers = [logging.getLogger(name) for name in logging.root.manager.loggerDict]
+for logger in loggers:
+    logger.setLevel(logging.WARNING)
+
+class PaceMakerJob(GenericJob):
+    def __init__(self, project, job_name):
+        super().__init__(project, job_name)
+        self.__name__ = "PaceMakerJob"
+        self.__version__ = "0.1"
+
+        self.input = GenericParameters(table_name="input")
+        self.input['cutoff'] = 10.
+        self.input['metadata'] = {}
+        self.input['data'] = {}  # data_config
+        self.input['potential'] = {}  # potential_config
+        self.input['fit'] = {}  # fit_config
+        self.input['backend'] = {'evaluator': 'tensorpot'}  # backend_config
+
+        self.structure_data = None
+
+        # self.executable = "pacemaker input.yaml -l log.txt"
+        self._executable = None
+        self._executable_activate()
+
+    def _save_structure_dataframe_pckl_gzip(self, df):
+        df.rename(columns={"number_of_atoms": "NUMBER_OF_ATOMS",
+                           "energy": "energy_corrected",
+                           "atoms": "ase_atoms"}, inplace=True)
+        df["NUMBER_OF_ATOMS"] = df["NUMBER_OF_ATOMS"].astype(int)
+        if "pbc" not in df.columns:
+            df["pbc"] = df["ase_atoms"].map(lambda atoms: np.all(atoms.pbc))
+
+        data_file_name = os.path.join(self.working_directory, "df_fit.pckl.gzip")
+        logging.info("Saving training structures dataframe into {} with pickle protocol = 4, compression = gzip".format(
+            data_file_name))
+        df.to_pickle(data_file_name, compression="gzip", protocol=4)
+        return data_file_name
+
+    def write_input(self):
+        # prepare datafile
+        if self.structure_data is None:
+            raise ValueError(
+                "`structure_data` is none, but should be pd.DataFrame, TrainingContainer or valid pickle.gzip filename")
+        if isinstance(self.structure_data, pd.DataFrame):
+            logging.info("structure_data is pandas.DataFrame")
+            data_file_name = self._save_structure_dataframe_pckl_gzip(self.structure_data)
+            self.input["data"] = {"filename": data_file_name}
+        elif isinstance(self.structure_data, str):  # filename
+            if os.path.isfile(self.structure_data):
+                logging.info("structure_data is valid file path")
+                self.input["data"] = {"filename": self.structure_data}
+            else:
+                raise ValueError("Provided structure_data filename ({}) doesn't exists".format(self.structure_data))
```

Comment:
Nit: we're transitioning over to a different string format
```suggestion
                raise ValueError(f"Provided structure_data filename ({self.structure_data}) doesn't exists")
```

Reply:
done